### PR TITLE
Gradle 8 support

### PR DESCRIPTION
**Pull request by stewartfrancis**
- Adjusts API usage to remove some internal API usage that was deprecated in Gradle 7 and removed in Gradle 8.  There'll be more of this to do for Gradle 9, but I've not done that now.
- Updates tests to run on Gradle 7.6.1 and 8.3, derived from a commit from @ind1go
- Adds a new sample project which demonstrates how to build an OSGi bundle and repackage into a CICS bundle in the same module
- Adds a `cicsBundle` configuration which could be used to resolve the CICS bundle project by down-stream projects.  Does not add a `Component` so this could be easily published, but that would be a logical next
- Small rewrite of how the tasks are wired together to prevent eager task creation.

I did also notice that we had an unnecessary property on the package task, which I removed.  This technically breaks our API, so we may want to consider a major version bump (or just deprecating that property instead of removing it).  There are other things we might consider if doing a major version bump anyway, like the changes to how to register additional bundle part configuration that @KyeMaloy97 and I discussed (which I've now forgotten about entirely), and making the tasks usable independently of the project and extensions.

### CODE REVIEW

File: src/main/kotlin/com/ibm/cics/cbgp/DeployBundleTask.kt [hunk 1]
```diff
@@ -48,6 +48,11 @@ open class DeployBundleTask : DefaultTask() {
 			""".trimIndent()
 	}
 
+	init {
+		// Never skip the deploy task, even if nothing has changed, because we can't know whether the bundle needs redeploying in CICS.
+		outputs.upToDateWhen { false }
+	}
+
```
Comment (ind1go) [inline src/main/kotlin/com/ibm/cics/cbgp/DeployBundleTask.kt:55]: Do we need to decide on one place to have this modification? It's also in `BundlePlugin.kt`.

File: src/main/kotlin/com/ibm/cics/cbgp/PackageBundleTask.kt [hunk 2]
```diff
@@ -27,12 +27,6 @@ open class PackageBundleTask : Zip() {
 	@InputDirectory
 	val inputDirectory: DirectoryProperty = project.objects.directoryProperty()
 
-	/**
-	 * Set the zip archive file as a task output. This will be linked to the input of the deploy task.
-	 */
-	@OutputFile
-	val outputFile: RegularFileProperty = project.objects.fileProperty()
-
```
Comment (ind1go) [inline src/main/kotlin/com/ibm/cics/cbgp/PackageBundleTask.kt:35]: As discussed, can I suggest we make this PR compatible with the current version 1 train and release, then go to a v2 snapshot for removing this property and also the additional changes mentioned in the description.

File: src/test/groovy/com/ibm/cics/cbgp/GoldenPathTests.groovy [hunk 3]
```diff
@@ -285,9 +303,12 @@ class GoldenPathTests extends AbstractTest {
...
 		""".stripIndent()
 
-		runGradleAndSucceed([BundlePlugin.DEPLOY_TASK_NAME])
+		runGradleAndSucceed([BundlePlugin.DEPLOY_TASK_NAME], gradleVersion)
 
 		then:
 		checkBuildOutputFiles(["PROGDEF2.program"])
+
+		where:
+		gradleVersion << GradleVersions.GRADLE_VERSIONS
```
Comment (ind1go) [inline src/test/groovy/com/ibm/cics/cbgp/GoldenPathTests.groovy:312]: Seems fair. I was trying to find a way to test on all test versions without having to remember to add the where clause to every test when you make it.

Maybe (different PR) we can add an extension to Spock to check whether each spec's tests have `gradleVersion` as an input. I don't know if that's possible.

File: src/test/groovy/com/ibm/cics/cbgp/GradleVersions.groovy [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.ibm.cics.cbgp
+
+abstract class GradleVersions {
+
+    static List GRADLE_VERSIONS = ["7.6.1", "8.3"]
```
Comment (ind1go) [inline src/test/groovy/com/ibm/cics/cbgp/GradleVersions.groovy:5]: We state support for Gradle 5 in the README - should be testing there too.

**Review by ind1go — CHANGES_REQUESTED** [review]

**Comment by stewartfrancis** [comment]
I've reintroduced the old property, and deprecated it.  Also addressed your other review comments (thanks) @ind1go ready for a re-review now.

File: src/main/kotlin/com/ibm/cics/cbgp/DeployBundleTask.kt (hunk 1 above)
Comment (stewartfrancis) [reply src/main/kotlin/com/ibm/cics/cbgp/DeployBundleTask.kt:55]: You're right, we shouldn't need it in the plugin and I'll remove

Comment (stewartfrancis) [reply src/main/kotlin/com/ibm/cics/cbgp/DeployBundleTask.kt:55]: Have updated now

**Review by ind1go — APPROVED** [review]

**Pull request merged**